bench-tps/src/main.rs

```diff
@@ -19,43 +18,118 @@ use {
     solana_gossip::gossip_service::{discover_cluster, get_client, get_multi_client},
     solana_rpc_client::rpc_client::RpcClient,
     solana_sdk::{
-        commitment_config::CommitmentConfig, fee_calculator::FeeRateGovernor, pubkey::Pubkey,
+        commitment_config::CommitmentConfig,
+        fee_calculator::FeeRateGovernor,
+        pubkey::Pubkey,
+        signature::{Keypair, Signer},
         system_program,
     },
-    solana_streamer::socket::SocketAddrSpace,
+    solana_streamer::{socket::SocketAddrSpace, streamer::StakedNodes},
     std::{
-        collections::HashMap, fs::File, io::prelude::*, net::SocketAddr, path::Path, process::exit,
-        sync::Arc,
+        collections::HashMap,
+        fs::File,
+        io::prelude::*,
+        net::{IpAddr, SocketAddr},
+        path::Path,
+        process::exit,
+        sync::{Arc, RwLock},
     },
 };
 
 /// Number of signatures for all transactions in ~1 week at ~100K TPS
 pub const NUM_SIGNATURES_FOR_TXS: u64 = 100_000 * 60 * 60 * 24 * 7;
 
+fn find_node_activated_stake(
+    rpc_client: Arc<RpcClient>,
+    node_id: Pubkey,
+) -> Result<(u64, u64), ()> {
+    let vote_accounts = rpc_client.get_vote_accounts();
+    if let Err(error) = vote_accounts {
+        error!("Failed to get vote accounts, error: {}", error);
+        return Err(());
+    }
+
+    let vote_accounts = vote_accounts.unwrap();
+
+    let total_active_stake: u64 = vote_accounts
+        .current
+        .iter()
+        .chain(vote_accounts.delinquent.iter())
```

Comment:
I am not familiar with this business logic, why do we add the delinquent here?

Reply:
I thought total active stake includes delinquent nodes stake (by doing experiments primarily). And also here it is done this way https://github.com/solana-labs/solana/blame/5108350710b42f8bc2950395a067dc1d15213ba4/cli/src/cluster_query.rs#L1903

Reply:
Total stake is coming from `StakedNodes` structure https://github.com/solana-labs/solana/blob/master/streamer/src/streamer.rs#L30, @lijunwangs do you think it is supposed to take into account delinquent nodes stake?

Reply:
I am not sure, the current validator code does not include the delinquent stake I think. Can you use delinquent stake to do communications/vote? If so, it should be included, otherwise we probably should not include.

Reply:
If I understand the way it works in the TPU side, it looks like there is no filter on delinquent https://github.com/solana-labs/solana/blob/master/core/src/staked_nodes_updater_service.rs#L82
But I've asked on [discord](https://discord.com/channels/428295358100013066/977244255212937306/1076092681706557491).

Reply:
So it looks like although atm delinquent stakes are taken into account, they are going to be unstaked https://github.com/solana-labs/solana/issues/24302
Due to that, will change this code to take into account only normal nodes